sqlglot/dataframe/sql/session.py

```diff
@@ -104,8 +106,15 @@ def createDataFrame(
         sel_expression = exp.Select(**select_kwargs)
         return DataFrame(self, sel_expression)
 
+    def _optimize(
+        self, expression: exp.Expression, dialect: t.Optional[Dialect] = None
+    ) -> exp.Expression:
+        dialect = dialect or self.dialect
+        quote_identifiers(expression, dialect=dialect)
```

Comment:
Why did we need to quote here prior to optimizing? Quotes are also added by the optimizer.

Reply:
Some tests fail if I remove it. This is likely covering up a bigger underlying issue that I will look into later.